spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -224,4 +224,130 @@
       }.to raise_error(WhitehallImporter::AbortImportError)
     end
   end
+
+  context "when the document is scheduled" do
+    let(:future_date) { Time.zone.now.tomorrow.rfc3339 }
+    let(:document) { create(:document, imported_from: "whitehall", locale: "en") }
+    let(:whitehall_edition) do
+      build(
+        :whitehall_export_edition,
+        state: "scheduled",
+        revision_history: [
+          build(:revision_history_event),
+          build(:revision_history_event, event: "update", state: "submitted"),
+          build(:revision_history_event, event: "update", state: "scheduled"),
+        ],
+        scheduled_publication: future_date,
+      )
+    end
+
+    let(:edition) do
+      described_class.call(document: document, whitehall_edition: whitehall_edition)
+    end
+
+    it "creates two statuses" do
+      expect(edition.statuses.count).to eq(2)
+    end
+
+    it "sets the correct scheduled status" do
+      expect(edition).to be_scheduled
+    end
+
+    it "sets the correct previous status" do
+      expect(edition.statuses.first).to be_submitted_for_review
+    end
```

Comment:
I'm not sure if these 3 things need to be separate tests, we can probably just have one test that captures that the state history is submitted_for_review then scheduled.